### ISSUE TITLE
Generated text makes no sense. Trying to auto-generate sentences like https://transformer.huggingface.co/doc/distil-gpt2

### ISSUE THREAD

**Issue by connecteev** [huggingface/transformers#1821]
I would like to replicate the text-generation behavior of https://transformer.huggingface.co/doc/distil-gpt2 on my MacOS. I went to that URL and typed "A doula is a person" and kept tabbing for auto-text generation, and got this:
```
A doula is a person who cares for a child or infant. A doula is not a midwife. But the two are not entirely independent. Why did I chose a doula and not myself? There are two good reasons: First, since my daughter was still growing , I had the luxury of time to learn as much knowledge about breastfeeding as I could. Second, I have the opportunity
```

Quite impressive. But now, I want to replicate this in Python on my system so I can auto-generate text that makes sense. I did the following:
1. Installed tensor flow
2. installed pytorch
3. installed huggingface :D
4.  Based on the recommendation in the [Examples section here](https://huggingface.co/transformers/examples.html#language-generation), I then ran:
```
python run_generation.py --model_type=gpt2 --model_name_or_path=gpt2
```

and gave it the same sample input:
```
Model prompt >>> A doula
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [00:01<00:00, 10.24it/s]
 hatter with a broad laugh and a guffaw over her stomach are eminently adorable. In
Model prompt >>> A doula is a person
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [00:02<00:00,  8.94it/s]
 who wishes to be free and independent while having her resources donated to society.

Gender Non-

```
As you can see, I get gibberish back, the sentences and context makes no sense. What am I missing?

**Comment by TheEdoardo93** [comment]
Have you ever tried to finetune some hyper-paremeters, such as **temperature** or **seed**? You can also modify the **length** of the generated text by OpenAI GPT-2.

_Tip_: if you want to make OpenAI GPT-2 model more comfortable and sure about its results, please set the temperature value lower than 0,5 (e.g. 0,2 or 0,3).

> I would like to replicate the text-generation behavior of https://transformer.huggingface.co/doc/distil-gpt2 on my MacOS. I went to that URL and typed "A doula is a person" and kept tabbing for auto-text generation, and got this:
> 
> ```
> A doula is a person who cares for a child or infant. A doula is not a midwife. But the two are not entirely independent. Why did I chose a doula and not myself? There are two good reasons: First, since my daughter was still growing , I had the luxury of time to learn as much knowledge about breastfeeding as I could. Second, I have the opportunity
> ```
> 
> Quite impressive. But now, I want to replicate this in Python on my system so I can auto-generate text that makes sense. I did the following:
> 
> 1. Installed tensor flow
> 2. installed pytorch
> 3. installed huggingface :D
> 4. Based on the recommendation in the [Examples section here](https://huggingface.co/transformers/examples.html#language-generation), I then ran:
> 
> ```
> python run_generation.py --model_type=gpt2 --model_name_or_path=gpt2
> ```
> 
> and gave it the same sample input:
> 
> ```
> Model prompt >>> A doula
> 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [00:01<00:00, 10.24it/s]
>  hatter with a broad laugh and a guffaw over her stomach are eminently adorable. In
> Model prompt >>> A doula is a person
> 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [00:02<00:00,  8.94it/s]
>  who wishes to be free and independent while having her resources donated to society.
> 
> Gender Non-
> ```
> 
> As you can see, I get gibberish back, the sentences and context makes no sense. What am I missing?
